app/questionnaire/questionnaire_schema.py

```diff
@@ -150,46 +205,61 @@ def _get_questions_by_id(self) -> dict[str, list[ImmutableDict]]:
     def _get_answers_by_id(self) -> dict[str, list[ImmutableDict]]:
         answers_by_id = defaultdict(list)
 
-        for question_set in self._questions_by_id.values():
-            for question in question_set:
-                question_id = question["id"]
-                for answer in question["answers"]:
-                    answer_id = answer["id"]
-                    self._parent_id_map[answer_id] = question_id
-
-                    answers_by_id[answer["id"]].append(answer)
-                    for option in answer.get("options", []):
-                        detail_answer = option.get("detail_answer")
-                        if detail_answer:
-                            detail_answer_id = detail_answer["id"]
-                            answers_by_id[detail_answer_id].append(detail_answer)
-                            self._parent_id_map[detail_answer_id] = question_id
+        for question in self._get_flattened_questions():
+            question_id = question["id"]
+            for answer in question["answers"]:
+                answer_id = answer["id"]
+                self._parent_id_map[answer_id] = question_id
+
+                answers_by_id[answer["id"]].append(answer)
```

Comment:
Minor but either use the `answer_id` var on line 212 or `answer["id"]`

Reply:
Didn't touch since it was existing, only reshuffled but updated while I'm here.